android/src/main/java/com/iterable/reactnative/Serialization.java

```diff
@@ -198,6 +199,20 @@ static IterableConfig.Builder getConfigFromReadableMap(ReadableMap iterableConte
                 configBuilder.setLogLevel(logLevel);
             }
 
+            if(iterableContextJSON.has("dataRegion")) {
+                int dataRegion = iterableContextJSON.getInt("dataRegion");
```

Comment:
Although enums are represented as integers internally, in java its treated as class. So a direct mapping like this is not possible. [More reading](https://stackoverflow.com/a/5878986)

Reply:
Isnt the data region in the RN SDK represented as an integer though?

Reply:
Oh I see... In the Android SDK, we need a getEnum function to get the enum value from the string. Thus, in RN SDK, we should set the value of the data region to the endpoint.